src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs

```diff
@@ -125,68 +129,77 @@ private void ProcessArray(JArray jArray)
+                    // The "startup" logger is used in this class for startup related logs. The public logger is used
+                    // for all other logging after startup.
+
+                    ScriptApplicationHostOptions options = _configurationSource.HostOptions;
+                    string hostFilePath = Path.Combine(options.ScriptPath, ScriptConstants.HostMetadataFileName);
+                    JObject hostConfigObject = LoadHostConfig(hostFilePath);
+                    InitializeHostConfig(hostFilePath, hostConfigObject);
+                    string sanitizedJson = SanitizeHostJson(hostConfigObject);
+
+                    _logger.HostConfigApplied();
+
+                    // Do not log these until after all the configuration is done so the proper filters are applied.
+                    _logger.HostConfigReading(hostFilePath);
+                    _logger.HostConfigRead(sanitizedJson);
+
+                    return hostConfigObject;
+                }
             }
 
             private void InitializeHostConfig(string hostJsonPath, JObject hostConfigObject)
             {
-                // If the object is empty, initialize it to include the version and write the file.
-                if (!hostConfigObject.HasValues)
+                using (_metricsLogger.LatencyEvent(MetricEventNames.InitializeHostConfiguration))
                 {
-                    _logger.HostConfigEmpty();
+                    // If the object is empty, initialize it to include the version and write the file.
+                    if (!hostConfigObject.HasValues)
+                    {
+                        _logger.HostConfigEmpty();
 
-                    hostConfigObject = GetDefaultHostConfigObject();
-                    TryWriteHostJson(hostJsonPath, hostConfigObject);
-                }
+                        hostConfigObject = GetDefaultHostConfigObject();
+                        TryWriteHostJson(hostJsonPath, hostConfigObject);
+                    }
 
-                if (hostConfigObject["version"]?.Value<string>() != "2.0")
-                {
-                    throw new HostConfigurationException($"The {ScriptConstants.HostMetadataFileName} file is missing the required 'version' property. See https://aka.ms/functions-hostjson for steps to migrate the configuration file.");
+                    if (hostConfigObject["version"]?.Value<string>() != "2.0")
+                    {
+                        throw new HostConfigurationException($"The {ScriptConstants.HostMetadataFileName} file is missing the required 'version' property. See https://aka.ms/functions-hostjson for steps to migrate the configuration file.");
+                    }
                 }
             }
 
             internal JObject LoadHostConfig(string configFilePath)
             {
-                JObject hostConfigObject;
-                try
+                using (_metricsLogger.LatencyEvent(MetricEventNames.LoadHostConfiguration))
```

Comment:
same here, is it useful to have breakdown for each method?

Reply:
This does a File.ReadAllText and a parse. Thought it might be useful to have this in addition to LoadHostConfigurationSource to know the difference in total time against i/o time 

---
In reply to: [333798607](https://github.com/Azure/azure-functions-host/pull/4973#discussion_r333798607) [](ancestors = 333798607)